app/models/export.rb

```diff
@@ -0,0 +1,88 @@
+# Adapter to fire off the correct job, track transitions and responses.
+class Export < ApplicationRecord
+  class UnknownExportTypeError < StandardError; end
+
+  include Informed
+
+  belongs_to :snap_application
+  validates :status, inclusion: { in: %i(new queued in_process succeeded failed
+                                         unnecessary) }
+  validates :destination, inclusion: { in: %i(fax email) }
+  before_validation { self.destination = destination.to_sym }
```

Comment:
always happy to avoid callbacks -- any way we can move these outside of `before_validation` ?

Reply:
Yeah... I think Rails 5 has types for attributes? can we make Status a symbol?

Reply:
are you talking about using AR:Enum? http://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html